src/gf_base.h

```diff
@@ -74,9 +74,7 @@ Field<T>::Field(T p, int n, bool calculate_root)
 }
 
 template <typename T>
-Field<T>::~Field()
-{
-}
+Field<T>::~Field() = default;
```

Comment:
minor remark: it could be declared inside the class directly, same for the others